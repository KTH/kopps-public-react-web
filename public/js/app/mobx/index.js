/* eslint no-use-before-define: ["error", "nofunc"] */

import MobxStoreProvider from './MobxStoreProvider'
import useStore from './useStore'
import { compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument } from './compress'
// import { compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument } from './compressWithPako'

export { MobxStoreProvider, useStore, compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument }
