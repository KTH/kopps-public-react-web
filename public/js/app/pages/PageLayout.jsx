import React from 'react'
import { Col, Row } from 'reactstrap'
import MainMenu from '../components/MainMenu'

function MainContent({ children }) {
  return <main id="mainContent">{children}</main>
}

function PageLayout({ children }) {
  return (
    <Row style={{ padding: '30px 0' }}>
      <MainMenu />
      <Col>
        <MainContent>{children}</MainContent>
      </Col>
    </Row>
  )
}

export default PageLayout
