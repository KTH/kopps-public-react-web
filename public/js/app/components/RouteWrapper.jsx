/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import { Route } from 'react-router-dom'
import PropTypes from 'prop-types'
import { MobxStoreProvider } from '../mobx'

function RouteWrapper({
  createBreadcrumbs,
  component: Component,
  layout: Layout,
  applicationStore,
  createMenuData,
  ...rest
}) {
  const menuData = createMenuData(applicationStore)
  const breadcrumbs = createBreadcrumbs(applicationStore)
  return (
    <MobxStoreProvider initCallback={() => applicationStore}>
      <Route
        {...rest}
        render={() => (
          <Layout breadcrumbs={breadcrumbs} menuData={menuData}>
            <Component />
          </Layout>
        )}
      />
    </MobxStoreProvider>
  )
}

RouteWrapper.propTypes = {
  createBreadcrumbs: PropTypes.func,
  component: PropTypes.oneOfType([PropTypes.node, PropTypes.func, PropTypes.elementType]).isRequired,
  layout: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  createMenuData: PropTypes.func.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  applicationStore: PropTypes.object.isRequired,
}

RouteWrapper.defaultProps = {
  createBreadcrumbs: () => ({ include: 'student', items: [] }),
}

export default RouteWrapper
