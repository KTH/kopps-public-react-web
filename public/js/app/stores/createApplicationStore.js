/* eslint no-use-before-define: ["error", "nofunc"] */

// @ts-check

// eslint-disable-next-line no-unused-vars
import { observable } from 'mobx'
import commonStore from './commonStore'
import curriculumStore from './curriculumStore'
import studyProgrammeStore from './studyProgrammeStore'
import searchCoursesStore from './searchCoursesStore'
import appendix1Store from './appendix1Store'
import appendix2Store from './appendix2Store'
import literatureStore from './literatureStore'

export default createApplicationStore

const store = {
  /**
   * @property {string} message
   */
  message: 'Hallo',
  /**
   * @method
   * @param {string} text
   */
  setMessage,
  /**
   * @property {string} lastAdmissionTerm
   */
  lastAdmissionTerm: '',
  /**
   * @method
   * @param {string} lastAdmissionTerm
   */
  setLastAdmissionTerm,
  /**
   * @method
   * @param {map<string, {}>} programmes
   */
  setProgrammes,

  /**
   * @property {map<string, {}>} programmes
   */
  programmes: [],

  /**
   * @method
   * @param {[]} currentSchoolsWithDepartments
   */
  setCurrentSchoolsWithDepartments,

  /**
   * @property {[]} currentSchoolsWithDepartments
   */
  currentSchoolsWithDepartments: [],

  /**
   * @method
   * @param {[]} deprecatedSchoolsWithDepartments
   */
  setDeprecatedSchoolsWithDepartments,

  /**
   * @property {[]} deprecatedSchoolsWithDepartments
   */
  deprecatedSchoolsWithDepartments: [],
  /**
   * @property {string} departmentName
   */
  departmentName: '',
  /**
   * @method
   * @param {string} departmentName
   */
  setDepartmentName,

  /**
   * @property {[]} departmentCourses
   */
  departmentCourses: [],
  /**
   * @method
   * @param {[]} departmentCourses
   */
  setDepartmentCourses,
  /**
   * @property {string} programmeName
   */
  programmeName: '',
  /**
   * @method
   * @param {string} programmeName
   */
  setProgrammeName,
  /**
   * @property {string} programmeCode
   */
  programmeCode: '',
  /**
   * @method
   * @param {string} programmeCode
   */
  setProgrammeCode,
  /**
   * @property {[]} programmeTerms
   */
  programmeTerms: [],
  /**
   * @method
   * @param {[]} programmeTerms
   */
  setProgrammeTerms,
  /**
   * @property {number} lengthInStudyYears
   */
  lengthInStudyYears: 0,
  /**
   * @method
   * @param {number} lengthInStudyYears
   */
  setLengthInStudyYears,
  /**
   * @property {string} thisHostBaseUrl
   */
  thisHostBaseUrl: null,
}

function createApplicationStore(storeId) {
  switch (storeId) {
    case 'curriculum':
      return {
        ...commonStore,
        ...curriculumStore,
      }
    case 'searchCourses':
      return {
        ...commonStore,
        ...searchCoursesStore,
      }
    case 'objectives':
    case 'extent':
    case 'eligibility':
    case 'implementation':
      return {
        ...commonStore,
        ...studyProgrammeStore,
      }
    case 'appendix1':
      return {
        ...commonStore,
        ...studyProgrammeStore,
        ...appendix1Store,
      }
    case 'appendix2':
      return {
        ...commonStore,
        ...studyProgrammeStore,
        ...appendix2Store,
      }
    case 'literatureList':
      return {
        ...commonStore,
        ...literatureStore,
      }
    default:
      return {
        ...commonStore,
        ...store,
      }
  }
}

function setMessage(text = 'Happy coding!! :)') {
  this.message = text
}

function setProgrammes(programmes) {
  this.programmes = [...programmes]
}

function setCurrentSchoolsWithDepartments(currentSchoolsWithDepartments) {
  this.currentSchoolsWithDepartments = currentSchoolsWithDepartments
}

function setDeprecatedSchoolsWithDepartments(deprecatedSchoolsWithDepartments) {
  this.deprecatedSchoolsWithDepartments = deprecatedSchoolsWithDepartments
}

function setDepartmentName(departmentName) {
  this.departmentName = departmentName
}

function setDepartmentCourses(departmentCourses) {
  this.departmentCourses = departmentCourses
}

function setLastAdmissionTerm(lastAdmissionTerm) {
  this.lastAdmissionTerm = lastAdmissionTerm
}

function setProgrammeName(programmeName) {
  this.programmeName = programmeName
}

function setProgrammeCode(programmeCode) {
  this.programmeCode = programmeCode
}

function setProgrammeTerms(programmeTerms) {
  this.programmeTerms = programmeTerms
}

function setLengthInStudyYears(lengthInStudyYears) {
  this.lengthInStudyYears = lengthInStudyYears
}
