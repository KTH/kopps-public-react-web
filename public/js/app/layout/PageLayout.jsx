import React, { useEffect } from 'react'
import ReactDOM from 'react-dom'
import { Col, Row } from 'reactstrap'
import { Breadcrumbs } from '@kth/kth-reactstrap/dist/components/utbildningsinfo'

import { useStore } from '../mobx'
import MainMenu from '../components/MainMenu'

function renderBreadcrumbsIntoKthHeader(language) {
  const breadcrumbContainer = document.getElementById('breadcrumbs-header')
  if (breadcrumbContainer) {
    ReactDOM.render(<Breadcrumbs include="directory" language={language} />, breadcrumbContainer)
  }
}

function MainContent({ children }) {
  return (
    <Col>
      <main id="mainContent">{children}</main>
    </Col>
  )
}

function PageLayout({ children }) {
  const { language } = useStore()
  useEffect(() => renderBreadcrumbsIntoKthHeader(language))
  return (
    // Container in publicLayout.handlebars – begin
    <Row>
      <MainMenu />
      <MainContent>{children}</MainContent>
    </Row>
    // Container – end
  )
}

export default PageLayout
