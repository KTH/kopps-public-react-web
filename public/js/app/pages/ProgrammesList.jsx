import React from 'react'
import PropTypes from 'prop-types'
import { Col, Row } from 'reactstrap'
import { PageHeading, LinkList, Link } from '@kth/kth-reactstrap/dist/components/studinfo'
import { CollapseDetails } from '@kth/kth-reactstrap/dist/components/utbildningsinfo'

import Lead from '../components/Lead'
import FooterContent from '../components/FooterContent'

import { useStore } from '../mobx'
import translate from '../../../../domain/translate'
import { programmeLink } from '../util/links'
import { formatShortTerm } from '../../../../domain/term'
import Article from '../components/Article'

function Heading({ size, text, id }) {
  switch (size) {
    case 'h1':
      return <h1 id={id}>{text}</h1>
    case 'h2':
      return <h2 id={id}>{text}</h2>
    case 'h4':
      return <h4 id={id}>{text}</h4>
    case 'h5':
      return <h5 id={id}>{text}</h5>
    case 'h6':
      return <h6 id={id}>{text}</h6>
    default:
      return <h3 id={id}>{text}</h3>
  }
}

function Section({ programmeType, children }) {
  const id = `progGroup${programmeType}`
  const ariaLabelledBy = `heading-${programmeType}`
  return (
    <section id={id} aria-labelledby={ariaLabelledBy}>
      {children}
    </section>
  )
}

function CurrentProgrammeDescription({ programme }) {
  const { language } = useStore()
  const t = translate(language)
  const { credits, creditUnitAbbr, firstAdmissionTerm } = programme
  const formattedTerm = formatShortTerm(firstAdmissionTerm, language)
  return <>{`, ${credits} ${creditUnitAbbr}, ${t('programmes_admitted_from')} ${formattedTerm}`}</>
}

function ObsoleteProgrammeDescription({ programme }) {
  const { language } = useStore()
  const t = translate(language)
  const { credits, creditUnitAbbr, firstAdmissionTerm, lastAdmissionTerm } = programme
  const formattedLastTerm = formatShortTerm(lastAdmissionTerm, language)
  if (firstAdmissionTerm) {
    const formattedFirstTerm = formatShortTerm(firstAdmissionTerm, language)
    return (
      <>{`, ${credits} ${creditUnitAbbr}, ${t('programmes_admitted')} ${formattedFirstTerm}â€“${formattedLastTerm}`}</>
    )
  }
  return <>{`, ${credits} ${creditUnitAbbr}, ${t('programmes_admitted_until')} ${formattedLastTerm}`}</>
}

function ProgrammesListItem({ programme, variant }) {
  const { language } = useStore()
  const { programmeCode, title } = programme
  return (
    <>
      <Link href={programmeLink(programmeCode, language)}>{`${title} (${programmeCode})`}</Link>
      {variant === 'obsolete' ? (
        <ObsoleteProgrammeDescription programme={programme} />
      ) : (
        <CurrentProgrammeDescription programme={programme} />
      )}
    </>
  )
}

function CurrentProgrammesList({ programmes }) {
  return (
    <LinkList>
      {programmes.map(programme => (
        <ProgrammesListItem key={programme.title} programme={programme} />
      ))}
    </LinkList>
  )
}

function ObsololeteProgrammesList({ programmeType, programmes = [] }) {
  if (!programmes.length) return null
  const { language } = useStore()
  const t = translate(language)
  const title = `${t('programmes_older')} (${t('programme_type')[programmeType]})`
  return (
    <CollapseDetails title={title}>
      <LinkList>
        {programmes.map(programme => (
          <ProgrammesListItem key={programme.title} variant="obsolete" programme={programme} />
        ))}
      </LinkList>
    </CollapseDetails>
  )
}

function ProgrammesList() {
  const { language, programmes } = useStore()
  const t = translate(language)
  return (
    <>
      <Row>
        <Col>
          <PageHeading>{t('programmes_list_header')}</PageHeading>
          <Lead text={t('programmes_list_lead')} />
        </Col>
      </Row>
      <Row>
        <Col>
          <Article classNames={['paragraphs']}>
            {programmes.map(programme => (
              <Section key={programme[0]} programmeType={programme[0]}>
                <Heading id={`heading-${programme[0]}`} size="h2" text={t('programme_type')[programme[0]]} />
                <CurrentProgrammesList programmes={programme[1].first} />
                <ObsololeteProgrammesList programmeType={programme[0]} programmes={programme[1].second} />
              </Section>
            ))}
          </Article>
        </Col>
      </Row>
      <Row>
        <Col>
          <FooterContent />
        </Col>
      </Row>
    </>
  )
}

// TODO: Cleanup prop types: consolidate and rename

Heading.propTypes = {
  size: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']).isRequired,
  text: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
}

Section.propTypes = {
  programmeType: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
}

CurrentProgrammeDescription.propTypes = {
  programme: PropTypes.shape({
    credits: PropTypes.number.isRequired,
    creditUnitAbbr: PropTypes.string.isRequired,
    firstAdmissionTerm: PropTypes.string.isRequired,
  }).isRequired,
}

ObsoleteProgrammeDescription.propTypes = {
  programme: PropTypes.shape({
    credits: PropTypes.number.isRequired,
    creditUnitAbbr: PropTypes.string.isRequired,
    firstAdmissionTerm: PropTypes.string,
    lastAdmissionTerm: PropTypes.string.isRequired,
  }).isRequired,
}

ProgrammesListItem.propTypes = {
  programme: PropTypes.shape({
    programmeCode: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  }).isRequired,
  variant: PropTypes.string,
}

ProgrammesListItem.defaultProps = {
  variant: '',
}

CurrentProgrammesList.propTypes = {
  programmes: PropTypes.arrayOf(
    PropTypes.shape({
      programmeCode: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    })
  ).isRequired,
}

ObsololeteProgrammesList.propTypes = {
  programmeType: PropTypes.string.isRequired,
  programmes: PropTypes.arrayOf(
    PropTypes.shape({
      programmeCode: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    })
  ),
}

ObsololeteProgrammesList.defaultProps = {
  programmes: [],
}

export default ProgrammesList
