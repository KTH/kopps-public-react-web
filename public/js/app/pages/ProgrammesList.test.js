/* eslint-disable react/prop-types */

import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { axe, toHaveNoViolations } from 'jest-axe'
import { StaticRouter } from 'react-router'
import { MobxStoreProvider } from '../mobx'

import ProgrammesList from './ProgrammesList'
import RouteWrapper from '../components/RouteWrapper'
import PageLayout from '../layout/PageLayout'

import createApplicationStore from '../stores/createApplicationStore'
import getMenuData from '../config/menuData'

import commonSettings from '../config/mocks/mockCommonSettings'

expect.extend(toHaveNoViolations)

const applicationStore = createApplicationStore()

const testProgrammes = {
  sv: [
    [
      'TARKU',
      {
        first: [
          {
            programmeCode: 'ARKIT',
            title: 'Arkitektutbildning',
            titleOtherLanguage: 'Degree Programme in Architecture',
            firstAdmissionTerm: '20072',
            credits: 300,
            creditUnitLabel: 'Högskolepoäng',
            creditUnitAbbr: 'hp',
            educationalLevel: 'BASIC',
            lengthInStudyYears: 5,
            owningSchoolCode: 'Arkitektur och samhällsbyggnad',
            degrees: [{ code: 'TARKU' }],
          },
        ],
        second: [
          {
            programmeCode: 'A',
            title: 'Arkitektutbildning',
            titleOtherLanguage: 'Degree Programme in Architecture',
            lastAdmissionTerm: '20071',
            credits: 270,
            creditUnitLabel: 'Högskolepoäng',
            creditUnitAbbr: 'hp',
            lengthInStudyYears: 5,
            owningSchoolCode: 'Arkitektur och samhällsbyggnad',
            degrees: [{ code: 'TARKU' }],
          },
        ],
      },
    ],
  ],
  en: [
    [
      'TARKU',
      {
        first: [
          {
            programmeCode: 'ARKIT',
            title: 'Degree Programme in Architecture',
            titleOtherLanguage: 'Arkitektutbildning',
            firstAdmissionTerm: '20072',
            credits: 300,
            creditUnitLabel: 'Credits',
            creditUnitAbbr: 'hp',
            educationalLevel: 'BASIC',
            lengthInStudyYears: 5,
            owningSchoolCode: 'Arkitektur och samhällsbyggnad',
            degrees: [{ code: 'TARKU' }],
          },
        ],
        second: [
          {
            programmeCode: 'A',
            title: 'Degree Programme in Architecture',
            titleOtherLanguage: 'Arkitektutbildning',
            lastAdmissionTerm: '20071',
            credits: 270,
            creditUnitLabel: 'Credits',
            creditUnitAbbr: 'hp',
            lengthInStudyYears: 5,
            owningSchoolCode: 'Arkitektur och samhällsbyggnad',
            degrees: [{ code: 'TARKU' }],
          },
        ],
      },
    ],
  ],
}

const WrapperProgrammesList = ({ lang }) => {
  applicationStore.setLanguage(lang)
  applicationStore.setBrowserConfig(commonSettings)

  const updatedApplicationStore = {
    ...applicationStore,
  }
  return (
    <StaticRouter>
      <RouteWrapper
        exact
        path="/student/kurser/org"
        createBreadcrumbs={() => ({ include: 'directory' })}
        component={ProgrammesList}
        layout={PageLayout}
        applicationStore={updatedApplicationStore}
        createMenuData={store => ({ selectedId: 'ProgrammesList', ...getMenuData(store) })}
      />
    </StaticRouter>
  )
}

const ProgrammesListWithLayout = ({ lang }) => {
  applicationStore.setLanguage(lang)
  applicationStore.setBrowserConfig(commonSettings)
  applicationStore.setProgrammes(testProgrammes[lang])
  const menuData = getMenuData(applicationStore)

  const updatedApplicationStore = {
    ...applicationStore,
  }
  return (
    <StaticRouter>
      <MobxStoreProvider initCallback={() => updatedApplicationStore}>
        <PageLayout breadcrumbs={{ include: 'directory' }} menuData={{ selectedId: 'ProgrammesList', ...menuData }}>
          <ProgrammesList />
        </PageLayout>
      </MobxStoreProvider>
    </StaticRouter>
  )
}

describe('Render component ProgrammesList within RouterWrapper', () => {
  test('simple render', done => {
    render(<WrapperProgrammesList lang="en" />)
    done()
  })
})

describe('Render component ProgrammesList within Layout', () => {
  test('verify that page is accessible', async () => {
    const { container } = render(<ProgrammesListWithLayout lang="en" />)
    expect(await axe(container)).toHaveNoViolations()
  })
  test('get page header in English', () => {
    render(<ProgrammesListWithLayout lang="en" />)
    const h1Header = screen.getByRole('heading', { level: 1 })
    expect(h1Header).toHaveTextContent('Courses Part of Programme')
  })
  test('get page header in Swedish', () => {
    render(<ProgrammesListWithLayout lang="sv" />)
    const h1Header = screen.getByRole('heading', { level: 1 })
    expect(h1Header).toHaveTextContent('Kurser inom program')
  })
  test('match to snapshot in English', done => {
    const { asFragment } = render(<ProgrammesListWithLayout lang="en" />)
    expect(asFragment()).toMatchSnapshot()
    done()
  })
  test('match to snapshot in Swedish', done => {
    const { asFragment } = render(<ProgrammesListWithLayout lang="sv" />)
    expect(asFragment()).toMatchSnapshot()
    done()
  })
})

describe('Render component ProgrammesList and check its menu, content and links', () => {
  test('check all links on the page in English', () => {
    render(<ProgrammesListWithLayout lang="en" />)
    const h2Header = screen.getByRole('heading', { level: 2 })
    expect(h2Header).toHaveTextContent('Master of Architecture')
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(9)
    // Menu links
    expect(links[0]).toHaveTextContent('Student at KTH')
    expect(links[0].href).toStrictEqual('http://localhost/student/?l=en')
    expect(links[1]).toHaveTextContent('Courses Part of Programme')
    expect(links[1].href).toStrictEqual('http://localhost/student/kurser/kurser-inom-program')
    expect(links[2]).toHaveTextContent('Search course')
    expect(links[2].href).toStrictEqual('http://localhost/student/kurser/sokkurs')
    expect(links[3]).toHaveTextContent('Courses by school')
    expect(links[3].href).toStrictEqual('http://localhost/student/kurser/org')
    expect(links[4]).toHaveTextContent('Study Handbook 00/01 to 07/08')
    expect(links[4].href).toStrictEqual('http://localhost/student/program/shb')
    // Main content links
    expect(links[5]).toHaveTextContent('Degree Programme in Architecture (ARKIT)')
    expect(links[5].href).toStrictEqual('http://localhost/student/kurser/program/ARKIT?l=en')
    expect(links[6]).toHaveTextContent('Degree Programme in Architecture (A)')
    expect(links[6].href).toStrictEqual('http://localhost/student/kurser/program/A?l=en')
    // Footer links
    expect(links[7]).toHaveTextContent('Central study counseling')
    expect(links[7].href).toStrictEqual('https://www.kth.se/en/studies/master/general-study-counselling-1.621634')
    expect(links[8]).toHaveTextContent('kopps@kth.se')
    expect(links[8].href).toStrictEqual('mailto:kopps@kth.se')
  })
  test('check all links on the page in Swedish', () => {
    render(<ProgrammesListWithLayout lang="sv" />)
    const h2Header = screen.getByRole('heading', { level: 2 })
    expect(h2Header).toHaveTextContent('Arkitektutbildning')
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(9)
    // Menu links
    expect(links[0]).toHaveTextContent('Student på KTH')
    expect(links[0].href).toStrictEqual('http://localhost/student/')
    expect(links[1]).toHaveTextContent('Kurser inom program')
    expect(links[1].href).toStrictEqual('http://localhost/student/kurser/kurser-inom-program')
    expect(links[2]).toHaveTextContent('Sök kurs')
    expect(links[2].href).toStrictEqual('http://localhost/student/kurser/sokkurs')
    expect(links[3]).toHaveTextContent('Kurser per skola')
    expect(links[3].href).toStrictEqual('http://localhost/student/kurser/org')
    expect(links[4]).toHaveTextContent('Studiehandboken 00/01 tom 07/08')
    expect(links[4].href).toStrictEqual('http://localhost/student/program/shb')
    // Main content links
    expect(links[5]).toHaveTextContent('Arkitektutbildning (ARKIT)')
    expect(links[5].href).toStrictEqual('http://localhost/student/kurser/program/ARKIT')
    expect(links[6]).toHaveTextContent('Arkitektutbildning (A)')
    expect(links[6].href).toStrictEqual('http://localhost/student/kurser/program/A')
    // Footer links
    expect(links[7]).toHaveTextContent('Central studievägledning')
    expect(links[7].href).toStrictEqual('https://www.kth.se/utbildning/traffakth/studievagledning')
    expect(links[8]).toHaveTextContent('kopps@kth.se')
    expect(links[8].href).toStrictEqual('mailto:kopps@kth.se')
  })
  test('get page content in English', () => {
    render(<ProgrammesListWithLayout lang="en" />)
    const content = screen.getByText(
      'Choose a programme below to get an overview of courses and study years forming the programme.'
    )
    expect(content).toBeInTheDocument()
  })
  test('get page content in Swedish', () => {
    render(<ProgrammesListWithLayout lang="sv" />)
    const content = screen.getByText(
      'Välj program nedan för att få en översikt av de kurser och årskursindelning som utgör programmet.'
    )
    expect(content).toBeInTheDocument()
  })
})
