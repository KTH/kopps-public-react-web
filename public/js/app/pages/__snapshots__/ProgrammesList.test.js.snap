// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component ProgrammesList within Layout match to snapshot in English 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      aria-labelledby="kth-local-navigation-title--mobile"
      class="kth-local-navigation--mobile"
    >
      <button
        class="kth-button menu"
        id="kth-local-navigation-title--mobile"
      >
        <span>
          Course and programme directory
        </span>
      </button>
      <dialog
        class="kth-mobile-menu left"
      >
        <div
          class="kth-mobile-menu__navigation"
        >
          <button
            class="kth-icon-button close"
          >
            <span
              class="kth-visually-hidden"
            >
              Close
            </span>
          </button>
        </div>
        <div
          class="mobile-menu__content"
        >
          <a
            class="kth-button back"
            href="/student/studier/?l=en"
          >
            Studies
          </a>
          <h2>
            Course and programme directory
          </h2>
          <ul>
            <li>
              <a
                href="/student/kurser/kurser-inom-program"
              >
                Programme Syllabuses
              </a>
            </li>
            <li>
              <a
                href="/student/kurser/sokkurs"
              >
                Search courses
              </a>
            </li>
            <li>
              <a
                href="/student/kurser/sokkurs-ny-design"
              >
                Search courses (new)
              </a>
            </li>
            <li>
              <a
                href="/student/kurser/org"
              >
                Courses by school
              </a>
            </li>
            <li>
              <a
                href="/student/program/shb"
              >
                Studies before 07/08
              </a>
            </li>
          </ul>
        </div>
      </dialog>
    </nav>
    <nav
      aria-labelledby="local-navigation-title"
      class="kth-local-navigation col"
      id="mainMenu"
    >
      <a
        class="kth-button back"
        href="/student/studier/?l=en"
      >
        Studies
      </a>
      <h2
        id="local-navigation-title"
      >
        Course and programme directory
      </h2>
      <ul>
        <li>
          <a
            href="/student/kurser/kurser-inom-program"
          >
            Programme Syllabuses
          </a>
        </li>
        <li>
          <a
            href="/student/kurser/sokkurs"
          >
            Search courses
          </a>
        </li>
        <li>
          <a
            href="/student/kurser/sokkurs-ny-design"
          >
            Search courses (new)
          </a>
        </li>
        <li>
          <a
            href="/student/kurser/org"
          >
            Courses by school
          </a>
        </li>
        <li>
          <a
            href="/student/program/shb"
          >
            Studies before 07/08
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="col"
      id="mainContent"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <header
            role="presentation"
          >
            <h1>
              Programme Syllabuses
            </h1>
          </header>
          <div
            class="lead"
          >
            <p>
              This is where you will find all programme syllabuses at KTH. Each year group within a programme has its own programme syllabus. The programme syllabus contains, among other things, information about which courses are included in the programme and what applies for selection and admission.
            </p>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <form
            aria-label="Programme-Syllabuses"
            class="form-group"
            name="searchForm"
            role="search"
            style="display: flex; align-items: end;"
          >
            <div>
              <label
                for="pattern"
              >
                Search programme syllabus
              </label>
              <input
                data-testid="pattern"
                id="pattern"
                maxlength="80"
                name="pattern"
                size="50"
                type="text"
                value=""
              />
            </div>
          </form>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <article
            class="article article--no-start-margin "
          >
            <section
              aria-labelledby="heading-TARKU"
              id="progGroupTARKU"
            >
              <h2
                id="heading-TARKU"
              >
                Master of Architecture
              </h2>
              <ul
                class="link-list"
              >
                <li>
                  <a
                    href="/student/kurser/program/ARKIT?l=en"
                  >
                    Degree Programme in Architecture (ARKIT)
                  </a>
                  , 300 hp, admitted/batch from Autumn 07
                </li>
              </ul>
              <details
                class=""
              >
                <summary
                  aria-label=""
                  class="white"
                >
                  Programmes without new admission (Master of Architecture)
                </summary>
                <div>
                  <ul
                    class="link-list"
                  >
                    <li>
                      <a
                        href="/student/kurser/program/A?l=en"
                      >
                        Degree Programme in Architecture (A)
                      </a>
                      , 270 hp, admitted/batch until Spring 07
                    </li>
                  </ul>
                </div>
              </details>
            </section>
          </article>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <footer
            class="border-top mt-4 pt-1"
            id="articleFooter"
          >
            <address>
              Contact for questions regarding education at KTH: 
              <a
                href="https://www.kth.se/en/studies/master/general-study-counselling-1.621634"
              >
                Central study counseling
              </a>
              <br />
              Contact regarding technical matters on this page: 
              <a
                href="mailto:kopps@kth.se"
              >
                kopps@kth.se
              </a>
            </address>
          </footer>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Render component ProgrammesList within Layout match to snapshot in Swedish 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      aria-labelledby="kth-local-navigation-title--mobile"
      class="kth-local-navigation--mobile"
    >
      <button
        class="kth-button menu"
        id="kth-local-navigation-title--mobile"
      >
        <span>
          Kurs- och programkatalogen
        </span>
      </button>
      <dialog
        class="kth-mobile-menu left"
      >
        <div
          class="kth-mobile-menu__navigation"
        >
          <button
            class="kth-icon-button close"
          >
            <span
              class="kth-visually-hidden"
            >
              Close
            </span>
          </button>
        </div>
        <div
          class="mobile-menu__content"
        >
          <a
            class="kth-button back"
            href="/student/studier/"
          >
            Studier
          </a>
          <h2>
            Kurs- och programkatalogen
          </h2>
          <ul>
            <li>
              <a
                href="/student/kurser/kurser-inom-program"
              >
                Utbildningsplaner
              </a>
            </li>
            <li>
              <a
                href="/student/kurser/sokkurs"
              >
                Sök kurser
              </a>
            </li>
            <li>
              <a
                href="/student/kurser/sokkurs-ny-design"
              >
                Sök kurser (ny)
              </a>
            </li>
            <li>
              <a
                href="/student/kurser/org"
              >
                Kurser per skola
              </a>
            </li>
            <li>
              <a
                href="/student/program/shb"
              >
                Studier före 07/08
              </a>
            </li>
          </ul>
        </div>
      </dialog>
    </nav>
    <nav
      aria-labelledby="local-navigation-title"
      class="kth-local-navigation col"
      id="mainMenu"
    >
      <a
        class="kth-button back"
        href="/student/studier/"
      >
        Studier
      </a>
      <h2
        id="local-navigation-title"
      >
        Kurs- och programkatalogen
      </h2>
      <ul>
        <li>
          <a
            href="/student/kurser/kurser-inom-program"
          >
            Utbildningsplaner
          </a>
        </li>
        <li>
          <a
            href="/student/kurser/sokkurs"
          >
            Sök kurser
          </a>
        </li>
        <li>
          <a
            href="/student/kurser/sokkurs-ny-design"
          >
            Sök kurser (ny)
          </a>
        </li>
        <li>
          <a
            href="/student/kurser/org"
          >
            Kurser per skola
          </a>
        </li>
        <li>
          <a
            href="/student/program/shb"
          >
            Studier före 07/08
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="col"
      id="mainContent"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <header
            role="presentation"
          >
            <h1>
              Utbildningsplaner
            </h1>
          </header>
          <div
            class="lead"
          >
            <p>
              Här hittar du alla utbildningsplaner på KTH. Varje årskull inom ett program har en egen utbildningsplan. I utbildningsplanen finns information om bland annat vilka kurser som ingår i programmet och vad som gäller för urval och behörighet.
            </p>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <form
            aria-label="Programme-Syllabuses"
            class="form-group"
            name="searchForm"
            role="search"
            style="display: flex; align-items: end;"
          >
            <div>
              <label
                for="pattern"
              >
                Sök utbildningsplan
              </label>
              <input
                data-testid="pattern"
                id="pattern"
                maxlength="80"
                name="pattern"
                size="50"
                type="text"
                value=""
              />
            </div>
          </form>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <article
            class="article article--no-start-margin "
          >
            <section
              aria-labelledby="heading-TARKU"
              id="progGroupTARKU"
            >
              <h2
                id="heading-TARKU"
              >
                Arkitektutbildning
              </h2>
              <ul
                class="link-list"
              >
                <li>
                  <a
                    href="/student/kurser/program/ARKIT"
                  >
                    Arkitektutbildning (ARKIT)
                  </a>
                  , 300 hp, antagna/kull fr.o.m. HT07
                </li>
              </ul>
              <details
                class=""
              >
                <summary
                  aria-label=""
                  class="white"
                >
                  Program utan nyantagning (Arkitektutbildning)
                </summary>
                <div>
                  <ul
                    class="link-list"
                  >
                    <li>
                      <a
                        href="/student/kurser/program/A"
                      >
                        Arkitektutbildning (A)
                      </a>
                      , 270 hp, antagna/kull t.o.m. VT07
                    </li>
                  </ul>
                </div>
              </details>
            </section>
          </article>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <footer
            class="border-top mt-4 pt-1"
            id="articleFooter"
          >
            <address>
              Kontakt för frågor om utbildning på KTH: 
              <a
                href="https://www.kth.se/studievagledning"
              >
                Central studievägledning
              </a>
              <br />
              Teknisk kontakt om denna sida: 
              <a
                href="mailto:kopps@kth.se"
              >
                kopps@kth.se
              </a>
            </address>
          </footer>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
